scripts/tools/DepsInfo.php

```diff
@@ -0,0 +1,333 @@
+table, th, td {
+  border: 1px solid black;
+  font-family: monospace;
+}
+.classlist {
+font-size: 12px;
+max-height: 150px;
+overflow-y: scroll;
+}
+.missed {
+color: darkred;
+}
+</style>';
+        $resultHtml .='<table><tr>
+<th>Extension ID</th>
+<th>Used Classes</th>
+<th>Dependencies in manifest</th>
+<th>Real dependencies</th>
+<th>Redundant (Mentioned in manifest but not used)</th>
+<th>Missed (Not mentioned in manifest but used in code)</th>
+<th>Cyclic dependencies</th>
+</tr>';
+        foreach ($result as $extId => $row) {
+            $classes = $row['classes'];
+            $resultHtml .= '<tr>
+                <td>'.$extId.'</td>
+                <td><div class="classlist">'.implode('<br>', $classes).'</div></td>
+                <td>'.implode('<br>', $row['manifestDeps']).'</td>
+                <td>'.implode('<br>', $row['realDeps']).'</td>
+                <td>'.implode('<br>', $row['redundantInManifest']).'</td>
+                <td><div class="missed">'.implode('<br>', $row['notMentionedInManifest']).'</div></td>
+                <td><div class="missed">'.implode('<br>', $row['cyclicDeps']).'</div></td>
+                </tr>';
+        }
+        $resultHtml .=  '</table>';
+
+        return $resultHtml;
+    }
+
+    /**
+     * @param DirectoryIterator $fileInfo
+     * @return mixed|null
+     */
+    private function getManifest(SplFileInfo $fileInfo)
+    {
+        $path = $fileInfo->getRealPath().DIRECTORY_SEPARATOR.'manifest.php';
+        if (file_exists($path)) {
+            return require $path;
+        }
+        return null;
+    }
+
+    /**
+     * @param SplFileInfo $fileInfo
+     * @return false|string
+     */
+    private function getComposer(SplFileInfo $fileInfo)
+    {
+        $path = $fileInfo->getRealPath().DIRECTORY_SEPARATOR.'composer.json';
+        if (file_exists($path)) {
+            return json_decode(file_get_contents($path), true);
+        }
+        return null;
+    }
+
+    /**
+     * @param DirectoryIterator $fileInfo
+     * @return string|null
+     */
+    private function getDependClasses(SplFileInfo $fileInfo)
+    {
+        $result = [];
+        $pathToDephpend = ROOT_PATH . 'vendor' . DIRECTORY_SEPARATOR
+            . 'dephpend' . DIRECTORY_SEPARATOR
+            . 'dephpend' . DIRECTORY_SEPARATOR
+            . 'bin' . DIRECTORY_SEPARATOR . 'dephpend';
+        $command = 'php ' . $pathToDephpend . ' text ' . $fileInfo;
+        $dependencies = shell_exec($command);
+        $dependenciesArray = preg_split("/\r\n|\n|\r/", $dependencies);
+        $dependenciesArray = array_filter($dependenciesArray);
+        foreach ($dependenciesArray as $dependency) {
+            $dependencyClass = explode(' --> ', $dependency);
+            $result[] = $dependencyClass[1];
+        }
+        return array_unique($result);
+    }
+
+    /**
+     * @param string $class
+     * @return mixed|null
+     */
+    private function classToExtensionId(string $class)
```

Comment:
`PHP Fatal error:  Uncaught TypeError: Argument 1 passed to oat\taoDevTools\scripts\tools\DepsInfo::classToExtensionId() must be of the type string, null given, called in {my_path}/tao/taoDevTools/scripts/tools/DepsInfo.php on line 132 and defined in {my_path}/tao/taoDevTools/scripts/tools/DepsInfo.php:281`